maisim/maisim.Game/Graphics/UserInterface/Overlays/WorkingBeatmap.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Collections.Generic;
+using maisim.Game.Beatmaps;
+using maisim.Game.Utils;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Logging;
+
+namespace maisim.Game.Graphics.UserInterface.Overlays
+{
+    /// <summary>
+    /// The object that use to get the information of the current beatmap that is being played.
+    /// </summary>
+    public class WorkingBeatmap : CompositeDrawable

```

Comment:
Not a huge fan of the way this currently works. Maybe could be a special bindable subclass instead if it is to be used to manage currently selected beatmap / difficulty game-wide?